### PR TITLE
APP-1629 : Create queue room

### PR DESCRIPTION
**Pull request by crepache**

### CODE REVIEW

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 1]
```diff
@@ -20,11 +20,22 @@
...
+import org.symphonyoss.client.exceptions.RoomException;
+import org.symphonyoss.client.model.Room;
+import org.symphonyoss.client.model.SymAuth;
+import org.symphonyoss.symphony.bots.helpdesk.bot.authentication.HelpDeskAuthenticationService;
 import org.symphonyoss.symphony.bots.helpdesk.bot.bootstrap.HelpDeskBootstrap;
+import org.symphonyoss.symphony.bots.helpdesk.bot.client.HelpDeskHttpClient;
+import org.symphonyoss.symphony.bots.helpdesk.bot.client.HelpDeskSymphonyClient;
+import org.symphonyoss.symphony.bots.helpdesk.bot.config.HelpDeskBotConfig;
 import org.symphonyoss.symphony.bots.helpdesk.bot.init.SpringHelpDeskBotInit;
+import org.symphonyoss.symphony.bots.helpdesk.bot.listener.HelpDeskRoomEventListener;
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Import not used.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 2]
```diff
@@ -41,6 +52,12 @@
   @Autowired
   private ApplicationContext applicationContext;
 
+  @Autowired
+  private HelpDeskSymphonyClient helpDeskSymphonyClient;
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: The autowired annotation is not required once you have been created this component outside the spring container.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 3]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
         "**/*.story", "**/excluded*.story");
   }
 
+  private Room getQueueRoom() {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Please include javadoc to explain the behavior of this method.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 4]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
...
+    try {
+      return createRoom(Boolean.TRUE);
+    } catch (RoomException e) {
+      try {
+        return createRoom(Boolean.FALSE);
+      } catch (RoomException e1) {
+        throw new IllegalStateException("Couldn't create queue room.");
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Please include the root cause.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 5]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
...
+      }
+    }
+  }
+
+  private Room createRoom(Boolean showHistory) throws RoomException {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Javadoc.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 6]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
...
+
+    HelpDeskAuthenticationService authService = new HelpDeskAuthenticationService(config, httpClient);
+    SymAuth symAuth = authService.authenticate();
+
+    helpDeskSymphonyClient = new HelpDeskSymphonyClient(httpClient);
+
+    try {
+      helpDeskSymphonyClient.init(symAuth, config.getEmail(), config.getAgentUrl(), config.getPodUrl());
+    } catch (InitException e) {
+      throw new IllegalStateException("Cannot instantiate symphony client.");
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Please include the root cause.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 7]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
...
+    symRoomAttributes.setViewHistory(showHistory);
+
+    String randomId = UUID.randomUUID().toString();
+    symRoomAttributes.setName("Queue Room" + randomId);
+    symRoomAttributes.setDescription("Queue Room" + randomId);
+
+    return helpDeskSymphonyClient.getRoomService().createRoom(symRoomAttributes);
+  }
+
+  private void initSymphonyClient() {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Javadoc.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 8]
```diff
@@ -108,4 +124,43 @@ public InjectableStepsFactory stepsFactory() {
...
+      }
+    }
+  }
+
+  private Room createRoom(Boolean showHistory) throws RoomException {
+    SymRoomAttributes symRoomAttributes = new SymRoomAttributes();
+    symRoomAttributes.setViewHistory(showHistory);
+
+    String randomId = UUID.randomUUID().toString();
+    symRoomAttributes.setName("Queue Room" + randomId);
```
Comment (robson-sanchez) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Please include the white space between the name and the UUID. The same comment for the description.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 1 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 2 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 5 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 3 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 7 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 8 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 6 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java (hunk 4 above)
Comment (crepache) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Done.

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**